### DISCUSSION TITLE
✅ 🚀 Feature Request: Half-star rating for Star Rating widget

### DISCUSSION THREAD

**Discussion by nu-ad** [elementor/elementor#24067] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

The current Elementor Star Rating widget lacks the ability to accurately depict rating of an element. Not all ratings can fit neatly within a whole number rating (0-5, 0-10). As such, there's a noticeable gap between a 4-star and a 5-star rating, leaving out a lot of nuanced feedback that can be captured in between.

### Describe the solution you'd like

I'd like the Elementor Star Rating widget to be enhanced to allow half-star ratings. This would provide a 0.5 increment, enabling users to rate something as 4.5 stars, for example, thereby providing a more accurate rating. This simple addition would enhance the granularity of user feedback without overcomplicating the system, as **it already display half ratings in the schema markup** and code itself.

### Describe alternatives you've considered

We've prepared a custom snippet for this, but why go around already invented widget such as the Star Ratings, as it only lacks this little feature that shouldn't be too hard to add (just another rule and an icon). 

### Additional context

The precision of feedback is crucial for website owners, product managers, and service providers. They use these ratings to gauge customer satisfaction and areas of improvement. Half-star ratings would provide a middle ground and a more detailed range of feedback, greatly benefiting businesses and users alike. Like I mentioned, the number is already there, all is required is just a little twitch to the widget code to add another icon for half ratings. 

![elementorhelloratingstarpls](https://github.com/elementor/elementor/assets/86675392/727288d7-97c5-4e3f-86d2-2d11d0669419)

**Comment by nu-ad** [comment]
I just noticed that the star rating widget actually shows a half-star, but only when the number is entered with decimal point, not a comma. 

I'm wondering, shouldn't that be treated as a bug? Both separators should be recognized.

**Comment by nicholaszein** [comment]
Hello, @nu-ad!

Thank you so much for your feedback! 🙏 

We have great news! 🙌 

### 📢 We're happy to announce that the request you raised was resolved with the release of the new Rating widget in Elementor v3.17! 🥳 

With this new widget, you have more options to choose the number of rating points, custom icons, and styling options, and it is fully compatible with Google Rich snippets (updated) and fully accessible. This also allows you to set decimal values, and even use Dynamic Tags to show your ratings.

Your current Start Rating widgets will remain as they were before this update. But moving forward, only the new Rating widget will be available to add to your layouts.

If you are using the old Star Rating widget, we would recommend that you replace it with the new Rating widget.

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
